### DISCUSSION TITLE
Do we have a function that does what the "Close with comment" in Github does?

### DISCUSSION THREAD

**Discussion by rodrigo-morales-1** [magit/forge#585] [Ideas]
As we all know,  the button "Close with comment" in Github, appears when commenting in those issues that are opened (see screenshot below). I want to have the same behavior using Forge within Emacs.

![send](https://github.com/magit/forge/assets/74389646/c543ba74-1554-4269-809c-99763e6a0963)

Assuming that I'm in a buffer whose major mode is `forge-topic-mode`, the following are the steps that I'm currently using to accomplish the same result of the aforementioned button:

1. Use `forge-create-post` (by default, bound to `C-c C-n`) to publish a new comment and publish it.
2. Move the cursor to the line that shows the state of the issue and call `forge-edit-topic-state` (by default, bound to `C-c C-e`)

I wish I didn't have to move my cursor and, desirably, that Emacs only sends one request instead of two.

I can spend some time writing a Elisp function, but before writing it, I want to make sure that we don't have a function for that.

**Comment by tarsius** [comment]
> 2. Move the cursor to the line that shows the state of the issue and call `forge-edit-topic-state` (by default, bound to `C-c C-e`)
>
> I wish I didn't have to move my cursor and, desirably, that Emacs only sends one request instead of two.

You can avoid having to move there by adding a binding in `forge-topic-mode-map`.

You will still have to finish the reply with `C-c C-c` and then close the topic in a separate step. IMO, that is not a problem.

> I can spend some time writing a Elisp function, but before writing it, I want to make sure that we don't have a function for that.

We don't have a command for that and I encourage you to write it, but I cannot guarantee that I will merge it.

**Reply by tarsius** [reply]
In topic buffers you can now type `C-<return>` to bring up the "do something to this topic" menu. Thus closing a topic is now a very quick action and I see even less reason to add a "add comment and close in one step" command.
